### ISSUE TITLE
Helm reports wrong managedFields manager

### ISSUE THREAD

**Issue by mattfarina** [helm/helm#9859]
Kubernetes 1.18 added something called `managedFields`. These are parts of the `metadata` section. Helm reports the manager for its part as `Go-http-client`. This can be found by using `helm create` to generate a chart, installing the chart, and then look at the fields with kubectl.

For example,

```console
❯ kubectl get deploy example --show-managed-fields -o json | jq ".metadata.managedFields[].manager"
"Go-http-client"
"k3s"
```

In this case the `managedFields` has details added by the kube pkgs from Helm and by k3s. The ones added by Helm (through the kube pkgs) are reported as `Go-http-client`.

Note, `--show-managed-fields` is only available in kubectl v1.21+.

Marking this as a bug as this should have been added as part of the update to 1.21 kube libraries. It's not something I would have expected someone to easily pickup in the update.

Output of `helm version`:

```
version.BuildInfo{Version:"v3.6+unreleased", GitCommit:"fa558f7843e9a3ce317754b121f5ecd2a505b109", GitTreeState:"dirty", GoVersion:"go1.16.4"}
```

This is code on main as this shows up in previous versions of Helm.

Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:52:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2+k3s1", GitCommit:"5a67e8dc473f8945e8e181f6f0b0dbbc387f6fca", GitTreeState:"clean", BuildDate:"2021-06-21T20:52:44Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): N/A (k3s in my example)

**Comment by mattfarina** [comment]
Note, there is a video on the `managedFields` feature at https://www.youtube.com/watch?v=P12z8X-qhis

**Comment by mattfarina** [comment]
This was fixed with the merging of #9870.
